include/boost/beast/core/detail/post.hpp

```diff
@@ -0,0 +1,106 @@
+//
+// Copyright (c) 2016-2017 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_DETAIL_POST_HPP
+#define BOOST_BEAST_DETAIL_POST_HPP
+
+#include <boost/asio/async_result.hpp>
+#include <boost/asio/executor_work_guard.hpp>
+#include <boost/asio/post.hpp>
+
+namespace boost {
+namespace beast {
+namespace detail {
+
+template<class Executor, class Handler>
+class post_op
+{
+public:
+    using executor_type =
+        boost::asio::associated_executor_t<Handler, Executor>;
+
+private:
+    boost::asio::executor_work_guard<Executor> wg1_;
+    boost::asio::executor_work_guard<executor_type> wg2_;
+    Handler h_;
+
+public:
+    post_op(post_op&&) = default;
```

Comment:
Redundant declarations?

Reply:
it isn't redundant, but I don't know if this file should even be there at all